File: 0344-reverse-string/0344-reverse-string.java
class Solution {
    public void reverseString(char[] s) {
        int st = 0;
        int ed = s.length-1;
        while(st<=ed){
            swap(s,st,ed);
            st++;
            ed--;
        }
    }
    
    private static void swap(char[] s, int i, int j){
        char temp = s[i];
        s[i] = s[j];
        s[j] = temp;
    }
}

File: 0344-reverse-string/0344-reverse-string.js
/**
 * @param {character[]} s
 * @return {void} Do not return anything, modify s in-place instead.
 */
var reverseString = function(s) {
    let start = 0;
    let end = s.length-1;
    while(start<=end){
        swap(s,start,end);
        start++;
        end--;
    }
    function swap(s,i,j){
        let temp = s[i];
        s[i] = s[j];
        s[j] = temp;
    }
};